common/src/main/java/com/commercetools/sunrise/payment/service/PaymentAdapterServiceImpl.java

```diff
@@ -0,0 +1,72 @@
+package com.commercetools.sunrise.payment.service;
+
+import com.commercetools.sunrise.payment.domain.CreatePaymentDataProvider;
+import com.commercetools.sunrise.payment.domain.PaymentServiceProvider;
+import io.sphere.sdk.payments.PaymentMethodInfo;
+import io.sphere.sdk.payments.PaymentStatus;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.ServiceLoader;
+
+/**
+ * Provides access to the payment service provider handling methods.
+ *
+ * Created by mgatz on 7/18/16.
+ */
+public class PaymentAdapterServiceImpl implements PaymentAdapterService {
+
+    private ServiceLoader<PaymentServiceProvider> pspLoader;
+
+    public PaymentAdapterServiceImpl() {
+
+        pspLoader = ServiceLoader.load(PaymentServiceProvider.class);
```

Comment:
Put a class implementing the PaymentServiceProvider interface on the classpath. Thats it. It will then be auto-discovered without needing any more configuration.

This is the way different payment modules could easily be added (or removed): add JAR to classpath, restart app, finished